src/handlers/rentals.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable @typescript-eslint/ban-types */
+/* eslint-disable prefer-const */
+import { log } from '@graphprotocol/graph-ts'
+import { BigInt } from '@graphprotocol/graph-ts'
+import { buildCountForAllRentals, buildCountForRentals } from '../modules/count'
+import { Rental } from '../entities/schema'
+import { OperatorUpdated, RentalStarted, AssetClaimed } from '../entities/Rentals/Rentals'
+
+function buildRentalId(contractAddress: string, tokenId: BigInt, index: BigInt): string {
+  return contractAddress + ':' + tokenId.toString() + ':' + index.toString()
+}
+
+export function handleRentalStart(event: RentalStarted): void {
+  let contractAddress = event.params._contractAddress.toHexString()
+  let tokenId = event.params._tokenId
+
+  let rentalIndex = buildCountForRentals(contractAddress, tokenId)
+  let newRentalId = buildRentalId(contractAddress, tokenId, rentalIndex.value)
+
+  let rental = new Rental(newRentalId)
+  rental.contractAddress = contractAddress
+  rental.tokenId = tokenId
+  rental.lessor = event.params._lessor.toHexString()
+  rental.tenant = event.params._tenant.toHexString()
+  rental.operator = event.params._operator.toHexString()
+  rental.rentalDays = event.params._rentalDays
+  rental.pricePerDay = event.params._pricePerDay
+  rental.sender = event.params._sender.toHexString()
+  rental.ownerHasClaimedAsset = false
+  rental.startedAt = event.block.timestamp
+  rental.save()
+  rentalIndex.save()
+
+  let metric = buildCountForAllRentals()
+  metric.save()
+}
+
+export function handleOperatorUpdated(event: OperatorUpdated): void {
+  let contractAddress = event.params._contractAddress.toHexString()
+  let tokenId = event.params._tokenId
+  let rentalIndex = buildCountForRentals(contractAddress, tokenId)
+  // Get the current one
+  rentalIndex.value.minus(BigInt.fromI32(1))
```

Comment:
I see you address this here. I believe it would be better to have some sort of getCountForRental instead of depending on the build and substracting

Reply:
I think it would be better to have utils for `getLastestRentalId(contractAddress,tokenId)`  `bumpRentalId(contractAddress,tokenId)`